Mathlib/Topology/Instances/EReal.lean

```diff
@@ -171,6 +170,182 @@ theorem tendsto_nhds_bot_iff_real {α : Type*} {m : α → EReal} {f : Filter α
   nhds_bot_basis.tendsto_right_iff.trans <| by simp only [true_implies, mem_Iio]
 #align ereal.tendsto_nhds_bot_iff_real EReal.tendsto_nhds_bot_iff_real
 
+/-! ### Liminfs and Limsups -/
+
+theorem liminf_le_liminf {α : Type*} {f : Filter α} {u v : α → EReal} (h : u ≤ᶠ[f] v) :
+    liminf u f ≤ liminf v f := Filter.liminf_le_liminf h
+
+theorem limsup_le_limsup {α : Type*} {f : Filter α} {u v : α → EReal} (h : u ≤ᶠ[f] v) :
+    limsup u f ≤ limsup v f := Filter.limsup_le_limsup h
+
+theorem limsup_add_le_of_lt {α : Type*} {f : Filter α} {u v : α → EReal} {a b : EReal}
```

Comment:
This lemma is a little bit strange, because it is completely superseded by other lemmas below. Can you add a docstring to the lemma, with pointers to the better statements?

Reply:
@sgouezel: Even `limsup_add_bot_ne_top` is superseded by `limsup_add_le_add_limsup` (although deriving the former from the latter is slightly cumbersome), so `limsup_add_le_add_limsup` is really the main result we care about.

It was, however, more comfortable to write three lemmas instead of a giant proof of one lemma.

What would be the best option? I can see a few possibilities :

- Keeping the file this way.
- Add docstrings pointing toward the right statements
- Folding the two obsolete lemmas into the proof of the third.
- Marking `limsup_add_le_of_lt` (and maybe `limsup_add_bot_ne_top`) as private.

By the way, since `limsup_add_le_of_lt` is superseded in the end, I don't think it is valuable to try to optimize its hypotheses or conclusion -- cf. yesterday's thread with you and @j-loreaux.

Reply:
Marking the lemmas as private *and* adding docstrings explaining by which statements they are superseded would probably be the best option.

Reply:
Ok, I have just added the following docstring: 

```lean
/-- This lemma is superseded by `limsup_add_le_of_le` (weaker hypothesis) and
`limsup_add_lt_of_lt` (stronger thesis). -/
lemma limsup_add_le_of_lt {α : Type*} {f : Filter α} {u v : α → EReal} {a b : EReal}
    (ha : limsup u f < a) (hb : limsup v f < b) : limsup (u + v) f ≤ a + b := by
```

Reply:
Note that my message is saying to mark the lemmas as private *and* add docstrings.

Reply:
Yes, I know. I was isolating the docstring to refer to it in this conversation.

Reply:
Now I have pushed the private status and the other docstring.